### ISSUE TITLE
"request.original" returns an empty Request {} object

### ISSUE THREAD

**Issue by Daniel-Ozori** [primatejs/primate#100]
`request.original` always returns an empty ` Request {}` object.

Setting `request.body.parse` to false in primate.config.js gives the same result and also makes the `request.body` return an empty object.

**Comment by terrablue** [comment]
Hi, thanks. That's just how node shows it. Did you try accessing some of the methods on `request.original`? At least on node, it's a WHATWG facade. On bun and deno it's an actually WHATWG object.

**Comment by terrablue** [comment]
The current implementation is here: https://github.com/rcompat/rcompat/blob/master/packages/rcompat/http/node/Request.js, and it's not yet completely conforming to WHATWG's Request. If anything is missing or not working as expected, could you please file an issue there?

**Comment by Daniel-Ozori** [comment]
Thank you for the clarification.

I looked through the code but the original/plaintext body isn't explicitly stored on the Request object.
 
To get the original body from request.original you have to directly read the stream then call stringify() on it like this.

~~~
import { Readable } from "node:stream";
import { stringify } from "rcompat/streams";

export default {
    
    async post(request){
         const original_body = await stringify(Readable.toWeb(request.original.body))
    }
}
~~~

And this also can only been done when `resquest.body.parse` is false because calling the json() function from https://github.com/rcompat/rcompat/blob/master/packages/rcompat/http/node/Request.js closes the ReadbleStream.

So calling request.json() on rcompat makes the body inaccessible and plaintext body isn't explicitly stored for other uses.

**Comment by Daniel-Ozori** [comment]
Not really a primate issue but since they're related

**Comment by terrablue** [comment]
Yeah, that method would work, but as it's currently only possible to change `request.body.parse` to `false` for the whole application, I see how that can become an issue.

What we can do is have https://github.com/rcompat/rcompat/blob/master/packages/rcompat/http/shared/Body.js parse `text/plain` as well. This would come as an rcompat patch update, so we wouldn't even need a new Primate version.

Would that work for you? You could then simply access `request.body` if you need to process a plaintext body in some of your routes.

**Comment by terrablue** [comment]
`Request#text` and body parsing of plaintext landed in rcompat 0.9.x and will be automatically available for the current Primate.

For disabling body parsing individually I created https://github.com/primatejs/primate/issues/101, scheduled to land in 0.31.

**Comment by Daniel-Ozori** [comment]
Thank you for all your help.
